### PR TITLE
Remove rails upperbound and test against latest release and rails HEAD

### PR DESCRIPTION
**Pull request by KaanOzkan**
We can remove Rails upperbound and test against edge. This would help app developers to update to latest Rails without updating this gem.

Also currently without having any Rails version conditional code we are testing against every minor version. We can structure this better by testing against lowest version, latest release and edge.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,16 @@ rvm:
 
 gemfile:
   - gemfiles/rails-5-0.gemfile
-  - gemfiles/rails-5-1.gemfile
-  - gemfiles/rails-5-2.gemfile
+  - gemfiles/rails-latest-release.gemfile
+  - gemfiles/rails-edge.gemfile
+
+matrix:
+  exclude:
```
Comment (KaanOzkan) [inline .travis.yml]: rails-edge (6.0.0.beta2) only supports ruby 2.5.0 or newer.

**Comment by KaanOzkan** [comment]
@unixcharles got time for a review? You seemed to have done the latest Rails dependency work. 
Also @rafaelfranca? Looks like you have write access

**Comment by rafaelfranca** [comment]
I don't have. But @NathanPJF definitely has

**Comment by NathanPJF** [comment]
If you need someone to merge, I can do that. However, would someone with more ruby/rails context please review? I don't work in that realm 😄

**Comment by rafaelfranca** [comment]
Of course. This change doesn't have any potential to break anything on this library. It only allows applications using any Rails versions greater than 5.0 to use this gem. It also add tests to make sure it works with Rails 6.0.

File: payment_icons.gemspec [hunk 2]
```diff
@@ -16,6 +16,6 @@ Gem::Specification.new do |s|
   s.test_files = Dir['test/**/*']
 
   s.add_dependency 'frozen_record'
-  s.add_dependency 'rails', '>= 5.0', '< 6.0'
+  s.add_dependency 'rails', '>= 5.0'
```
Comment (rafaelfranca) [inline payment_icons.gemspec]: This should be done in a different PR but this gem should not be depending on the entire `rails` framework. It probably only need `railties`.

Comment (KaanOzkan) [reply payment_icons.gemspec]: Created https://github.com/activemerchant/payment_icons/issues/171 . I'll take a look.

**Comment by tauthomas01** [comment]
Hi @KaanOzkan ,

I have merged https://github.com/activemerchant/payment_icons/pull/172 and it results a conflict here. Could you take look?

File: gemfiles/rails-latest-release.gemfile [hunk 3]
```diff
@@ -2,4 +2,4 @@ source "https://rubygems.org"
 
 gemspec path: '..'
 
-gem 'rails', '~> 5.1.0'
+gem 'rails'
```
Comment (KaanOzkan) [inline gemfiles/rails-latest-release.gemfile]: @rafaelfranca I wanna confirm it makes sense to test against rails since the goal is to see how the gem will work against a rails application. And that this is better than testing against railties versions only

Comment (rafaelfranca) [reply gemfiles/rails-latest-release.gemfile]: 👍 It does make sense.
